alerta/auth/basic_ldap.py

```diff
@@ -40,11 +48,20 @@ def login():
     except Exception as e:
         raise ApiError(str(e), 500)
 
+    # Get email address from LDAP
+    if not email_verified:
+        try:
+            ldap_result = ldap_connection.search_s(userdn, ldap.SCOPE_SUBTREE,'(objectClass=*)',['mail'])
```

Comment:
How generic is this attempt to get the email address? Would this work for most LDAP implementations, just some, or only yours?

Reply:
I'm pretty sure it'll work for most. The 'mail' attribute seems to be the standard, can't find examples of anything else being used. Doesn't mean it'll contain a value of course.

I noticed that you need an email address to create an api key (and maybe elsewhere) which is why I put a placeholder in if the email can't be looked up. 

Can we not make the email address editable from the UI so that it can be updated if required?

Reply:
You only need a login to create an API key so the email placeholder is not strictly necessary. And yes we could make email editable but I tend not to allow editing user attributes that are sourced from elsewhere otherwise you get discrepancies that are difficult to resolve.